processes-services/src/main/java/org/exoplatform/processes/rest/ProcessesRest.java

```diff
@@ -204,4 +205,74 @@ public Response getWorks(@ApiParam(value = "Identity technical identifier", requ
+          @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error"), })
+  public Response createWork(@ApiParam(value = "Work object to create", required = true)
+                                     WorkEntity workEntity) {
+    if (workEntity == null) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("Work object is mandatory").build();
+    }
+    if (workEntity.getProjectId() == 0 && workEntity.getWorkFlow().getProjectId() == 0 ) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("Work projectId object is mandatory").build();
+    }
+    long currentIdentityId = RestUtils.getCurrentUserIdentityId(identityManager);
+    if (currentIdentityId == 0) {
+      return Response.status(Response.Status.UNAUTHORIZED).build();
+    }
+    try {
+      Work newWork = processesService.createWork(EntityBuilder.toWork(processesService,workEntity),currentIdentityId);
+      return Response.ok(EntityBuilder.toWorkEntity(processesService, newWork, "")).build();
+    } catch (IllegalAccessException e) {
+      LOG.warn("User '{}' attempts to create a Work Work", e);
+      return Response.status(Response.Status.UNAUTHORIZED).entity(e.getMessage()).build();
+    } catch (Exception e) {
+      LOG.warn("Error creating a Work", e);
+      return Response.serverError().entity(e.getMessage()).build();
+    }
+  }
+
+  @PUT
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  @RolesAllowed("users")
+  @Path("/works")
+  @ApiOperation(value = "Updates a new work", httpMethod = "PUT", response = Response.class, consumes = "application/json")
+  @ApiResponses(value = { @ApiResponse(code = HTTPStatus.NO_CONTENT, message = "Request fulfilled"),
+          @ApiResponse(code = HTTPStatus.BAD_REQUEST, message = "Invalid query input"),
+          @ApiResponse(code = HTTPStatus.UNAUTHORIZED, message = "Unauthorized operation"),
+          @ApiResponse(code = HTTPStatus.INTERNAL_ERROR, message = "Internal server error"), })
+  public Response updateWork(@ApiParam(value = "Work object to update", required = true)
+                                         WorkEntity workEntity) {
+    if (workEntity == null) {
+      return Response.status(Response.Status.BAD_REQUEST).entity("Work object is mandatory").build();
+    }
+    long currentIdentityId = RestUtils.getCurrentUserIdentityId(identityManager);
+    if (currentIdentityId == 0) {
+      return Response.status(Response.Status.UNAUTHORIZED).build();
+    }
+    try {
+      Work newWork = processesService.updateWork(EntityBuilder.toWork(processesService, workEntity),
+              currentIdentityId);
+      return Response.ok(EntityBuilder.toWorkEntity(processesService,newWork, "")).build();
+    } catch (ObjectNotFoundException e) {
+      LOG.debug("User '{}' attempts to update a not existing work workFlow '{}'", currentIdentityId, e);
+      return Response.status(Response.Status.NOT_FOUND).entity("Work workFlow not found").build();
```

Comment:
```suggestion
      return Response.status(Response.Status.NOT_FOUND).entity("Work workFlow not found").build();
```
To be added in list of api responeses